runtime/jvmti/jvmtiHelpers.c

```diff
@@ -720,6 +721,12 @@ getThreadState(J9VMThread *currentThread, j9object_t threadObject)
 		state |= JVMTI_THREAD_STATE_TERMINATED;
 	} else {		
 		state |= JVMTI_THREAD_STATE_ALIVE;
+		/* Based on the isSuspendedInternalOffset flag, set the thread state to suspended
+		 * for corrosponing thread.
+		 */
+		if (0 != J9OBJECT_U32_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset)) {
+			state |= JVMTI_THREAD_STATE_SUSPENDED;
+		}
```

Comment:
- added ifdef'ed to avoid impacting older JDKs.
- corrected typo error (corresponding).
- added another block to reset the JVMTI_THREAD_STATE_SUSPENDED flag
  based on the isSuspendedInternal field.
- edited comment